docs/NNNN-hash-and-id-n-tuple-storage-layout.md

```diff
@@ -0,0 +1,325 @@
+
+## Encapsulation Directory
+
+For basic OCFL object identifiers, the object identifier is used as the name of the encapsulation directory (ie. the object root directory).
+
+Some object identifiers could contain characters that are not safe for directory names on all filesystems. Safe characters are defined as A-Z, a-z, 0-9, '-' and '\_'. When an unsafe character is encountered in an object identifier, it is percent-encoded using the lower-case hex characters of its UTF-8 encoding.
+
+Some object identifiers could also result in an encoded string that is longer than can be supported as a directory name. To handle that scenario, if the percent-encoded object identifier is longer than 100 characters, it is truncated to 100 characters, and then the digest of the original object identifier is appended to the encoded object identifier like this: <encoded-object-identifier-first-100-chars>-<digest>.
+
+| Object ID | Encapsulation Directory Name |
+| --- | --- |
+| object-01 | object-01 |
+| ..Hor/rib:lè-$id | %2e%2eHor%2frib%3al%c3%a8-%24id |
+| abcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghija | abcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghij-5cc73e648fbcff136510e330871180922ddacf193b68fdeff855683a01464220 |
+
+## Parameters
+
+### Summary
+
+* **Name:** `digestAlgorithm`
+  * **Description:** The digest algorithm to apply to the OCFL object identifier; MUST be an algorithm that is allowed in the OCFL fixity block
+  * **Type:** string
+  * **Constaints:** Must not be empty
+  * **Default:** sha256
+* **Name**: `tupleSize`
+  * **Description:** Indicates the size of the segments (in characters) that the digest is split into
+  * **Type:** number
+  * **Constraints:** An integer between 0 and 32 inclusive
+  * **Default:** 3
+* **Name:** `numberOfTuples`
+  * **Description:** Indicates how many segments are used for path generation
+  * **Type:** number
+  * **Constraints:** An integer between 0 and 32 inclusive
+  * **Default:** 3
+
+### Details
+
+#### digestAlgorithm
+
+`digestAlgorithm` is defaulted to `sha256`, and it MUST either contain a digest algorithm that's [officially supported by the OCFL spec](https://ocfl.io/draft/spec/#digest-algorithms) or defined in a community extension. The specified algorithm is applied to OCFL object identifiers to produce hex encoded digest values that are then mapped to OCFL object root paths.
+
+#### tupleSize
+
+`tupleSize` determines the number of digest characters to include in each tuple. The tuples are used as directory names. The default value is `3`, which means that each directory in the OCFL storage hierarchy could contain up to 4096 files. Increasing this value increases the maximum number of files and sub-directories per directory.
```

Comment:
Leaf directories can have more than `16^tupleSize` entries. I wonder whether it might be better to change the text here to say "...each **intermediate** directory in the OCFL storage hierarchy could contain up to 4096 files."?

Reply:
thanks, I'll update